lib/matplotlib/contour.py

```diff
@@ -1586,70 +1611,48 @@ def _initialize_x_y(self, z):
     _contour_doc = """
         Plot contours.
 
+        Call signature::
+
+            contour([X, Y,] Z, [levels], **kwargs)
+
         :func:`~matplotlib.pyplot.contour` and
         :func:`~matplotlib.pyplot.contourf` draw contour lines and
         filled contours, respectively.  Except as noted, function
         signatures and return values are the same for both versions.
 
-        :func:`~matplotlib.pyplot.contourf` differs from the MATLAB
-        version in that it does not draw the polygon edges.
-        To draw edges, add line contours with
-        calls to :func:`~matplotlib.pyplot.contour`.
-
-
-        Call signatures::
-
-          contour(Z)
 
-        make a contour plot of an array *Z*. The level values are chosen
-        automatically.
-
-        ::
-
-          contour(X,Y,Z)
```

Comment:
Although this example list is long, I think it makes much more sense than the 1-liner that has replaced it above, so I would be keen to keep these lines.

Reply:
You mean the stuff before `Optional keyword arguments`? Comparing [old](https://matplotlib.org/api/_as_gen/matplotlib.axes.Axes.contour.html?highlight=contour#matplotlib.axes.Axes.contour) and [new](https://8691-1385122-gh.circle-artifacts.com/0/home/circleci/project/doc/build/html/api/_as_gen/matplotlib.axes.Axes.contour.html?highlight=contour#matplotlib.axes.Axes.contour), this is mostly repetitive stuff:

- Draw Z,
- Draw Z with X, Y
- Draw Z with parameter N
- Draw Z with X, Y and  parameter N
- Draw Z with parameter V (which is just a more explicit control of the same property controlled by V)
- Draw Z with X, Y and  parameter V (which is just a more explicit control of the same property controlled by V)

It's much easier to say and understand:

- Draw Z, optionally with X, Y and optionally with giving the levels.

One might add a sentence after the call signature, "where Z are the values, X, Y are the coordinates, ..". But then again, that's just a repetition of the parameter section, which is immediately following.

Reply:
Well by definition, the square brackets mean [optional](https://stackoverflow.com/a/21654227/2726279). You can use or leave out every square bracket separately. So, `X, Z` does not work, but `Z, levels` does.

It's maybe a question if everybody knows this convention. I'd rather use a standard python function signature instead. However that's not possible with optional positional parameters, which are often used in matplotlib. Anyway, I'm still in favor of using the standard convention for optional parameters compared to verbosely giving multiple alternatives.

I could add a sentence "X, Y must be given both, or left out both." to the paramter description, if that adds additional clarity.